inference/worker/work.py

```diff
@@ -40,9 +41,17 @@ def _prepare_message(message: inference.MessageRead) -> str:
     # Construct prompt
     messages = [_prepare_message(message) for message in work_request.thread.messages]
 
-    # Prepend system prompt if it was specified in work parameters
-    if work_request.parameters.system_prompt:
-        pre_prompt = V2_SYSTEM_PREFIX + work_request.parameters.system_prompt + eos_token
+    # Prepend system prompt and custom_instructions if it was specified in work parameters
+    if (
+        work_request.parameters.system_prompt
+        or work_request.parameters.user_profile
+        or work_request.parameters.user_response_instructions
+    ):
+        pre_prompt = f"""{V2_SYSTEM_PREFIX}{work_request.parameters.system_prompt if work_request.parameters.system_prompt else ""}\n{
+        CUSTOM_INSTRUCTIONS_PREFIX.format(
```

Comment:
Can we change this so that the whole custom instruction prefix is not prepended at all if no instructions are given? Otherwise we are just using up tokens for no reason, as the worker retains the system prompt in context always at the cost of other tokens

Reply:
It should be like that right now, At least in my env, while testing.

When there is not system prompt or instructions, nothing is prepended to
the prompt.

Reply:
Or you meant, that we need to have, ```work_request.parameters.system_prompt or (work_request.parameters.user_profile
        and work_request.parameters.user_response_instructions)```
        ?

Reply:
> Or you meant, that we need to have, `work_request.parameters.system_prompt or (work_request.parameters.user_profile and work_request.parameters.user_response_instructions)` ?

I mean, in the case where `system_prompt` is provided, but neither `user_profile` or `user_response_instructions` are provided, we will currently get a system prompt with the `USER_INSTRUCTIONS_PREFIX` included, right?

So my suggestion would be something like this (not tested, I just wrote this in GitHub editor):
```
    work_params = work_request.parameters
    if work_params.system_prompt or work_params.user_profile or work_params.user_response_instructions:
        pre_prompt = V2_SYSTEM_PREFIX + (work_params.system_prompt or "")

        if work_params.user_profile or work_params.user_response_instructions:
            pre_prompt = f"""{pre_prompt}\n{CUSTOM_INSTRUCTIONS_PREFIX.format(user_profile=work_params.user_profile or "", user_response_instructions=work_params.user_response_instructions or "")}"""

        pre_prompt = pre_prompt + eos_token
```
Unless I'm missing something here and the prefix already would be omitted somehow in that case?

Reply:
Thanks,
I will checkout this and commit changes.